### PR TITLE
Hw jv dao5

### PR DESCRIPTION
**Pull request by RuslanIbIsh**

### CODE REVIEW

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ManufacturerDaoJdbcImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturer (name, country, deleted) "
```
Comment (YuraBondarenko) [inline src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java:21]: you can set 'deleted' by default false when creating a table

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        String query = "SELECT * FROM manufacturer WHERE deleted = false AND id = ?";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            Manufacturer manufacturer = null;
+            while (resultSet.next()) {
+                manufacturer = constructManufaturerFromResultSet(resultSet);
+            }
+            return Optional.of(manufacturer);
```
Comment (YuraBondarenko) [inline src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java:54]:
```suggestion
            if (resultSet.next()) {
                return Optional.of(constructManufaturerFromResultSet(resultSet));
            }
```

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    @Override
+    public List<Manufacturer> getAll() {
+        String query = "SELECT * FROM manufacturer WHERE deleted = false";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Manufacturer> manufacturers = new ArrayList<>();
+            while (resultSet.next()) {
+                Manufacturer manufacturer = constructManufaturerFromResultSet(resultSet);
+                manufacturers.add(manufacturer);
```
Comment (YuraBondarenko) [inline src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java:69]:
```suggestion
                manufacturers.add(constructManufaturerFromResultSet(resultSet));
```

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+    @Override
+    public boolean delete(Long id) {
+        String query = "UPDATE manufacturer SET deleted=true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, id);
+            int result = preparedStatement.executeUpdate();
+            return result > 0;
```
Comment (YuraBondarenko) [inline src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java:105]:
```suggestion
            return preparedStatement.executeUpdate() > 0;
```

**Review by YuraBondarenko — CHANGES_REQUESTED** [review]

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 1 above)
Comment (YuraBondarenko) [inline src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java:21]: `String query`
make naming more informative in all methods

**Review by YuraBondarenko — CHANGES_REQUESTED** [review]

**Review by YuraBondarenko — APPROVED** [review]

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 1 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java]: rewrote

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 2 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java]: rewrote

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 3 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java]: rewrote

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 4 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java]: rewrote

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 1 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java]: rewrote

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String insertQuery = "INSERT INTO manufacturer (name, country) "
+                + "VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement =
+                        connection.prepareStatement(insertQuery, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, manufacturer.getName());
+            preparedStatement.setString(2, manufacturer.getCountry());
+            int result = preparedStatement.executeUpdate();
```
Comment (RomanDubovskyi) [inline src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java:28]:
```suggestion
           preparedStatement.executeUpdate();
```
we don't need it, if we can't add manufacturer, we will receive SQL exception

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getObject(1, Long.class));
+            }
+            if (result > 0) {
+                return manufacturer;
+            }
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't get connection in method create", throwables);
+        }
+        throw new DataProcessingException("Can't return manufacturer from method create");
```
Comment (RomanDubovskyi) [inline src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java:39]:
```suggestion
```
second throwing is redundant

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+            preparedStatement.setString(2, manufacturer.getCountry());
+            int result = preparedStatement.executeUpdate();
+            ResultSet resultSet = preparedStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                manufacturer.setId(resultSet.getObject(1, Long.class));
+            }
+            if (result > 0) {
+                return manufacturer;
+            }
+        } catch (SQLException throwables) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java:36]:
```suggestion
        } catch (SQLException e) {
```

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+            ResultSet resultSet = statement.executeQuery();
+            Manufacturer manufacturer = null;
+            while (resultSet.next()) {
+                return constructManufaturerFromResultSet(resultSet);
+            }
+
+        } catch (SQLException throwables) {
+            throw new DataProcessingException("Can't get connection in method get", throwables);
+        }
+        throw new DataProcessingException("Can't get Manufacturer in method get");
```
Comment (RomanDubovskyi) [inline src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java:57]:
```suggestion
        return Optional.empty();
```
if no object found we return empty optional

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 5 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java]: rewote

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 6 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java]: deleted

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 7 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java]: done

File: src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java (hunk 8 above)
Comment (RuslanIbIsh) [reply src/main/java/com/iri/dao3/dao/jdbc/ManufacturerDaoJdbcImpl.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**